File: Assignments/A2/assn2.py
import re

value = []
try:
    with open("a2_input.txt") as file:
        value = file.readlines()
except:
    print("Error: a2_input.txt file not found")

regex = re.compile(r'\$(\*)*(\d{0,2}(,\d{3})*|0).\d\d')

print("Starting Python Program Type to Check RegEx\n")

#ToDo: start verifying printing is only temporary
for i in range(len(value)):
    value[i] = value[i].replace('\n', '')
    if regex.match(value[i]):
        print(value[i] + " is a valid money format.")
    else:
        print(value[i] + " is an invalid money format.")

File: Assignments/A2/assn2.java
import java.io.File;
import java.util.*;

public class assn2{
    public static void main(String [] args){
        try{
            String value = "";
            String regex = "^\\$(\\*)*(\\d{0,2}(,\\d{3})*|0).\\d\\d";
            System.out.println("Starting Java Program Type to Check RegEx\n");
            //File file = new File("C:\\Users\\Brian Pham\\Desktop\\CSULB\\CECS 424 - Programming Languages\\Assignments\\A2\\a2_input.txt");
            File file = new File(args[0]);
            Scanner sc = new Scanner(file);
            while(sc.hasNext()){
                value = sc.next();
                if(value.matches(regex)){
                    System.out.println(value + " is a valid money format.");
                }
                else {
                    System.out.println(value + " is an invalid money format.");
                }
                //ToDo: start verifying printing is only temporary
            }

            sc.close();
        }
        catch(Exception e){
            System.out.println(e.getMessage());
        }
    }
}